kivy/uix/behaviors/compoundselection.py

```diff
@@ -308,7 +308,7 @@ def select_with_touch(self, node, touch=None):
             return False
 
         if (node in self.selected_nodes and (not range_select)):  # selected
-            if multiselect:
+            if ((not multi) or multiselect):
```

Comment:
Also in android or other touch devices , one has to set multiselect=True as well as touch_multiselect=True for getting the feature of multiselection of nodes.